src/clojure_lsp/crawler.clj

```diff
@@ -209,26 +209,25 @@
           (log/info "Manual GC after classpath scan took" (float (/ (- (System/nanoTime) start-time) 1000000000)) "seconds")
           (db/save-deps root-path project-hash classpath analysis))))))
 
-(defn ^:private analyze-project! [project-root]
-  (let [root-path (shared/uri->path project-root)
+(defn ^:private analyze-project! [project-root-uri]
+  (let [root-path (shared/uri->path project-root-uri)
         settings (:settings @db/db)
         source-paths (get settings :source-paths)]
     (analyze-classpath! root-path source-paths settings)
     (log/info "Analyzing source paths for project root" root-path)
     (analyze-paths! source-paths false)))
 
-(defn initialize-project [project-root client-capabilities client-settings]
-  (let [project-settings (config/resolve-config project-root)
-        root-path (shared/uri->path project-root)
-        settings (-> (merge client-settings project-settings)
+(defn initialize-project [project-root-uri client-capabilities]
+  (let [project-settings (config/resolve-config project-root-uri)
+        root-path (shared/uri->path project-root-uri)
+        settings (-> (merge (:settings @db/db) project-settings)
                      (update :source-paths (fn [source-paths] (mapv #(str (.getAbsolutePath (to-file root-path %))) source-paths)))
                      (update :cljfmt cljfmt.main/merge-default-options))]
     (when-let [log-path (:log-path settings)]
       (logging/update-log-path log-path))
     (swap! db/db assoc
-           :project-root project-root
+           :project-root-uri project-root-uri
            :project-settings project-settings
-           :client-settings client-settings
```

Comment:
Why remove the client settings? Without that we can surely know if a setting was setted via a client (lsp-mode, calva) or a user (.lsp/config.edn)

Reply:
Oh got it, you just did it sooner

Reply:
Hum, it's a little bit odd we add the settings the call before this function and then change it again on this function, can't we just make in a single place?

Reply:
I moved it to handlers/initialize where I am also setting the settings for the URI format.

Reply:
Yeah, my point is that we `swap!` those settings before `initialize-project` and right after calling it here, overriding the `:settings`, can't we do that logic in a single function in a single place to make it simpler?

Reply:
Yes, that would be better.

Reply:
I've move all the settings initialisation into the initialize-project function, similar to how it was before.